### PR TITLE
Add callback functions

### LINKED ISSUE #103: Add callbacks
* [x] Seek
 * [x] Debug
 * [x] Match
 * [x] Chunk begin
 * [x] Chunk end

### PR DESCRIPTION
**Pull request by moteus**
Close #103

### CODE REVIEW

File: src/lceasy.c [hunk 1]
```diff
@@ -1009,6 +1097,243 @@ static int lcurl_easy_set_PROGRESSFUNCTION(lua_State *L){
...
+  lcurl_easy_t *p = lcurl_geteasy(L);
+  return lcurl_easy_set_callback(L, p, &p->chunk_bgn,
+    CURLOPT_CHUNK_BGN_FUNCTION, CURLOPT_CHUNK_DATA,
+    "chunk_bgn", lcurl_chunk_bgn_callback
+  );
+}
+
+static int lcurl_easy_set_CHUNK_END_FUNCTION(lua_State *L){
+  lcurl_easy_t *p = lcurl_geteasy(L);
+  return lcurl_easy_set_callback(L, p, &p->chunk_bgn,
```
Comment (moteus) [inline src/lceasy.c]: Fix callback pointer

File: src/lceasy.c [hunk 2]
```diff
@@ -1200,6 +1560,17 @@ static const lcurl_const_t lcurl_easy_opt[] = {
...
+  // Debug message types not easy info
+  OPT_ENTRY(INFO_TEXT         )
+  OPT_ENTRY(INFO_HEADER_IN    )
+  OPT_ENTRY(INFO_HEADER_OUT   )
+  OPT_ENTRY(INFO_DATA_IN      )
+  OPT_ENTRY(INFO_DATA_OUT     )
+  OPT_ENTRY(INFO_SSL_DATA_OUT )
+  OPT_ENTRY(INFO_SSL_DATA_IN  )
+#undef OPT_ENTRY
+
```
Comment (moteus) [inline src/lceasy.c]: Add curl file types constants

File: src/lceasy.c [hunk 3]
```diff
@@ -1009,6 +1097,243 @@ static int lcurl_easy_set_PROGRESSFUNCTION(lua_State *L){
...
+  lua_newtable(L);
+  lua_pushstring (L, info->filename  ); lua_setfield(L, -2, "filename"   );
+  lua_pushinteger(L, info->filetype  ); lua_setfield(L, -2, "filetype"   );
+  lutil_pushint64(L, info->time      ); lua_setfield(L, -2, "time"       );
+  lutil_pushint64(L, info->perm      ); lua_setfield(L, -2, "perm"       );
+  lua_pushinteger(L, info->uid       ); lua_setfield(L, -2, "uid"        );
+  lua_pushinteger(L, info->gid       ); lua_setfield(L, -2, "gid"        );
+  lutil_pushint64(L, info->size      ); lua_setfield(L, -2, "size"       );
+  lutil_pushint64(L, info->hardlinks ); lua_setfield(L, -2, "hardlinks"  );
+  lutil_pushint64(L, info->flags     ); lua_setfield(L, -2, "flags"      );
```
Comment (moteus) [inline src/lceasy.c]: Convert flags to set?

File: src/lceasy.c [hunk 4]
```diff
@@ -1009,6 +1097,243 @@ static int lcurl_easy_set_PROGRESSFUNCTION(lua_State *L){
...
+    CURLOPT_CHUNK_BGN_FUNCTION, CURLOPT_CHUNK_DATA,
+    "chunk_bgn", lcurl_chunk_bgn_callback
+  );
+}
+
+static int lcurl_easy_set_CHUNK_END_FUNCTION(lua_State *L){
+  lcurl_easy_t *p = lcurl_geteasy(L);
+  return lcurl_easy_set_callback(L, p, &p->chunk_end,
+    CURLOPT_CHUNK_END_FUNCTION, CURLOPT_CHUNK_DATA,
+    "chunk_end", lcurl_chunk_bgn_callback
```
Comment (data-man) [inline src/lceasy.c]: Maybe lcurl_chunk_end_callback?

Comment (moteus) [inline src/lceasy.c]: Really need tests :)
But this is FTP only functions and I do not know how to install ftp on travis without sudo.

Comment (data-man) [inline src/lceasy.c]: Sorry, for the Windows the library not compiled without the Lake. :(

Comment (moteus) [inline src/lceasy.c]: There MSVC project or `luarocks make rockspecs/lua-curl-scm-0.rockspec`

**Pull request merged**